scanner/families/malware/yara/yara.go

```diff
@@ -91,7 +94,7 @@ func (s *Scanner) Run(sourceType utils.SourceType, userInput string) error {
 		defer cleanup()
 
 		// Define the yara args to run
-		args := []string{"-C", s.compiledRuleFile, "-r", "-w", "-m", fsPath}
```

Comment:
Couple of suggestions here:
- We can fully remove the _per file timeout_ field and use the context passed to `Run` to control the cancellation/timeout
- We dont need to scan each file independently but switch it to per-directory scan
- We actually dont need to parallelise the execution on our own but rather use `--threads=runtime.NumCPU()` to pass this responsibility to Yara since its supported on them
- Instead of running multiple processes parallely to scan each dir, we can do this sequentially since the parallelisation is supported on Yara
- Scanning a directory by Yara should use recursive flag `-r` to ensure that the whole directory depth is scanned